MaxText/max_utils.py

```diff
@@ -440,20 +440,42 @@ def get_kv_cache_annotations(model, config, rng, mesh):
   """ Get a shaped abstraction of the state (including optimizer)"""
 
   def init_kv_cache(model, config):
+    if config.combined_kv_cache:
```

Comment:
Oh interesting -- I think your point here is that model init was kind of wrong before? Because in either case we should be init-ing with the config.max_prefill_predict_length as input?

As it happens I think this isn't a big issue because the cache size is being driven by config constants not by init (so using the wrong shape is actually ok!)

Anyways, we should fix this and just run through with the `config.max_prefill_predict_length`?